### PR TITLE
Remove cmake dependency

### PR DESCRIPTION
**Pull request by xobs**
This is an experimental patch that removes the dependency on cmake.  It has been fully tested on Windows and Unix, and Darwin.

Additionally, a CI test for Darwin has been added, although the test currently times out on Travis in `src/controls/progressbar.rs - controls::progressbar::ProgressBarValue (line 19)`.  This is an upstream issue (see https://github.com/andlabs/libui/issues/495).

Platform support:

* [x] Darwin
* [x] Unix
* [x] Windows MSVC
* [x] Windows GNU

This has not been tested without the `build` feature enabled, as I am not familiar with how that should work.

### CODE REVIEW

**Comment by xobs** [comment]
I've added Unix support.

Note that this same approach could be taken to bypass https://github.com/rust-native-ui/libui-rs/issues/70 and synchronize with a newer version of libui without adding a dependency on python.

**Comment by xobs** [comment]
I've added Darwin support back in, but I don't have access to a machine to actually test it.

**Comment by xobs** [comment]
I've added a Travis CI target for Darwin, so I've proven that it actually builds.  Some tests run, but the actual test times out.

File: ui-sys/build.rs [hunk 1]
```diff
@@ -42,39 +51,259 @@ fn main() {
...
+            link("windowscodecs", false);
+        } else if unix {
+            base_config.include(format!("{}{}", src_base, "/unix"));
+
+            let pkg_cfg = pkg_config::Config::new().probe("gtk+-3.0").unwrap();
+            for inc in pkg_cfg.include_paths {
+                base_config.include(inc);
+            }
+
+            base_config.file(format!("{}{}", src_base, "/unix/alloc.c"));
```
Comment (MOZGIII) [inline ui-sys/build.rs]: Consider extracting `format!("{}{}", src_base, path)` into a helper function.

Comment (xobs) [reply ui-sys/build.rs]: Alright, I've done this.

**Comment by NoraCodes** [comment]
xobs, thanks for sticking with this! I'm currently looking for a Mac I can test this on, but it looks great.

**Comment by xobs** [comment]
@NoraCodes have you managed to find a Mac to test yet?

I suspect that bbab0fb73da0e55fbc512812e5d60a42ae9a5ea4 is caused by the Mac UI never posting the "Quit" message, and so should be a relatively easy fix.  But again, the problem is that I don't have access to a Mac right now so I can't test.

**Comment by xobs** [comment]
I've gone further and replaced the files with an iter of strings.

I've tested this new build system with the tip of libui as well, and it builds just fine.  So this approach allows libui-rs to take a newer version of libui.

Note that I had a version of the build system that actually parsed the `meson.build` files, but (1) the build got huge due to building `rustpython_parser`, and (2) it didn't work with e.g. `windows` due to the `windows = import("windows")` statement on top, which throws a token error.

**Comment by MOZGIII** [comment]
Iter if much better! Why not just glob files from the dir btw?

**Comment by xobs** [comment]
We definitely could glob files, and I'd prefer to do that instead of parsing the `meson.build` file.  However, I chose to explicitly list all files in case `libui` added some test code, or example code, or other files that are shipped but not used.

If you prefer I can refactor it to use a glob.  The current file list was taken by reading the `meson.build` file and doing `s/'/"/g`.

**Comment by xobs** [comment]
> 
> 
> Iter if much better! Why not just glob files from the dir btw?

For example, in `common/`, there is a file called `OLD_table.c` that should be ignored.  If we globbed the directory, this would be included.

**Comment by xobs** [comment]
I managed to get access to an OS X machine, and can confirm this patch works.  The only thing I downloaded was rust, and I downloaded git when prompted by the OS.  This was able to build and run.

As a separate issue, bbab0fb73da0e55fbc512812e5d60a42ae9a5ea4 is required because this patchset also includes OS X Travis tests, and it turns out that test never worked on Darwin due to the ordering of calls.  It turns out only Linux and Windows support calling Quit on a UI before entering its Main.

I'll raise an issue about this upstream.

**Comment by xobs** [comment]
Posted at https://github.com/andlabs/libui/issues/495

**Comment by xobs** [comment]
I updated the initial PR text.

I have since also verified that this works under `x86_64-pc-windows-gnu` in addition to `x86_64-pc-windows-msvc`.

**Comment by vaskoa** [comment]
O, man. You a wonderful human being @xobs.

After struggling with this for an entire day I can confirm it finally works for me: Windows + MSVC. 😂

**Comment by NoraCodes** [comment]
I'm tired of this being blocked. It works for me on Windows and Ubuntu; I haven't been able to test it out on Mac OS but I can't see a reason it wouldn't work. I'm inclined to merge this if @barzamin agrees.

**Comment by barzamin** [comment]
@NoraCodes I agree. I can try to find a macos machine to test on if you want

**Comment by xobs** [comment]
I did test it on a mac, and confirmed that it works, however the bug in testing exists there.  This is not a regression, however, as before this patch there weren't any mac tests.  I can understand you wanting to independently verify the claims, though.

**Comment by xobs** [comment]
Just an FYI (in case this ever gets merged): The failure in the doctests that I work around in the macOS code will become a failure for all platforms in the next release of libui: https://github.com/andlabs/libui/commit/faaccd0413f913fa06f218c461e20a761f568bc7 so that doctest will need to get rewritten for all platforms, and not call `uiQuit()` before calling `uiInit()`.

**Pull request merged**